tests/test_verify_base_url.py

```diff
@@ -54,3 +60,21 @@ def test_disable_verify_via_env(testdir, httpserver, monkeypatch):
     httpserver.serve_content(content='<h1>Error!</h1>', code=500)
     result = testdir.runpytest('--base-url', httpserver.url)
     assert result.ret == 0
+
+
+def test_verify_retries_url(testdir, httpserver, monkeypatch):
```

Comment:
This test doesn't appear to be doing anything new. If you want to test that we can have a URL that initially fails but recovers then you'll need to do it in a single `inline_run`. I'd suggest changing the httpserver response in a separate thread after a short delay. I was able to get this working using:

```python
def test_verify_retries_url(testdir, monkeypatch, httpserver):
    def slow_start(httpserver):
        httpserver.serve_content(content='<h1>Error!</h1>', code=500)
        time.sleep(1)
        httpserver.serve_content(content='<h1>Success!</h1>', code=200)
    testdir.makepyfile('def test_pass(): pass')
    monkeypatch.setenv('VERIFY_BASE_URL', True)
    monkeypatch.setattr(Retry, 'BACKOFF_MAX', 1)
    t = Thread(target=slow_start, args=(httpserver,))
    t.start()
    result = testdir.runpytest('--base-url', httpserver.url)
    assert result.ret == 0
```

I tested this by increasing the sleep to 10 seconds, which then caused the test to fail. I'd suggest seeing if you can assert stderr for the passing test to check for a line that matches '127.0.0.1 - - [25/May/2017 18:21:12] "GET / HTTP/1.1" 500 -' to ensure we recover from a 500 status code.